### PR TITLE
Converted to Use Shared DSCResources Code and CodeCov.io - Fixes #143

### LINKED ISSUE #143: Convert to use Shared AppVeyor Code and Add CodeCov.io
I'll submit a PR for this.

### PR DESCRIPTION
**Pull request by PlagueHO**
* Converted AppVeyor.yml to use DSCResource.tests shared code.
* Opted-In to markdown rule validation.
* Readme.md modified resolve markdown rule violations.
* Added CodeCov.io support.

---
This change is [Reviewable](https://reviewable.io/reviews/powershell/xactivedirectory/145)

### CODE REVIEW

**Comment by PlagueHO** [comment]
I think to get the CodeCov.io to work, a maintainer of this repo will need to enable it. Is this correct @TravisEz13 ?

File: appveyor.yml [hunk 1]
```diff
@@ -24,18 +21,8 @@ build: false
...
-            $env:PSModulePath = "$pwd;$env:PSModulePath"
-        }
-        $testResultsFile = ".\TestsResults.xml"
-        $res = Invoke-Pester -OutputFormat NUnitXml -OutputFile $testResultsFile -PassThru
-        (New-Object 'System.Net.WebClient').UploadFile("https://ci.appveyor.com/api/testresults/nunit/$($env:APPVEYOR_JOB_ID)", (Resolve-Path $testResultsFile))
-        if ($res.FailedCount -gt 0) {
-            throw "$($res.FailedCount) tests failed."
-        }
+        Invoke-AppveyorTestScriptTask `
+            -CodeCoverage
```
Comment (TravisEz13) [inline appveyor.yml]: You are missing the `-codeCovIo`

Comment (TravisEz13) [inline appveyor.yml]: We enabled CodeCov.io at the org level.  The missing code should be the only issue.

**Comment by PlagueHO** [comment]
@TravisEz13 - Doh! I've upgraded 4 of these repos now and completely forgot this! Fixed.

File: appveyor.yml [hunk 2]
```diff
@@ -3,15 +3,12 @@
...
-        Push-Location
-        cd DscResource.Tests
-        Import-Module .\TestHelper.psm1 -force
-        Pop-Location
-        Install-PackageProvider -Name NuGet -MinimumVersion 2.8.5.201 -Force
-        Install-Module -Name Pester -Repository PSGallery -Force
+    - git clone https://github.com/PowerShell/DscResource.Tests
+    - ps: |
         Add-WindowsFeature RSAT-AD-PowerShell
+        Import-Module "$env:APPVEYOR_BUILD_FOLDER\DscResource.Tests\AppVeyor.psm1"
```
Comment (TravisEz13) [inline appveyor.yml]: used named parameter

**Comment by TravisEz13** [comment]
@PlagueHO LGTM

**Comment by PlagueHO** [comment]
@kwirkykat - do you know who is maintaining this repo? It would be good to get this one merged :grin:

**Comment by johlju** [comment]
According to this [Maintainers.md](https://github.com/PowerShell/DscResources/blob/master/Maintainers.md#current-maintainers) it is @kwirkykat and @mbreakey3 😄

**Comment by mbreakey3** [comment]
LGTM

**Pull request merged**